### PR TITLE
Adding statement cleanup

### PR DESCRIPTION
**Pull request by Vesihiisi**
Clean up the reference adding mechanism in Monument.add_statement(), broken out from [comment](https://github.com/Vesihiisi/COH-tools/pull/47#discussion_r107470744).

- Add a single reference directly, without surrounding it with list markers.
- WLM_source used as default reference (when no reference given).
- Force not adding any reference with refs = False
- Use None instead of empty lists as function parameters.

Implement this in practice in SeArbetsl to see it works.

Task: https://phabricator.wikimedia.org/T161303

### CODE REVIEW

File: importer/Monument.py [hunk 1]
```diff
@@ -94,10 +94,18 @@ def print_wd_to_table(self):
...
+        Add a statement to the data object.
+
+        If the given property already exists, this will
+        append another value to the array,
         i.e. two statements with the same prop.
+
+        :param prop_name: name of the property, as stated in the props library file
+        :param value: the value of the property
+        :param quals: a dictionary of qualifiers
+        :param refs: a list of references
```
Comment (lokal-profil) [inline importer/Monument.py]: Describe the None/True/False behaviour here

File: importer/Monument.py [hunk 2]
```diff
@@ -108,6 +116,12 @@ def add_statement(self, prop_name, value, quals={}, refs=[]):
             for k in quals:
                 prop_name = self.props[k]
                 qualifiers[prop_name] = quals[k]
+        if refs is None or refs is True:
+            refs = self.wlm_source
+        elif refs is False:
+            refs = None
+        if refs and not isinstance(refs, list):
```
Comment (lokal-profil) [inline importer/Monument.py]: I would add a linebreak to separate this from the "pre-processing" if-else block

File: importer/Monument.py [hunk 3]
```diff
@@ -179,17 +191,18 @@ def set_coords(self, coord_keywords_tuple):
...
             else:
-                ref = self.wlm_source
                 self.add_statement(
-                    "coordinates", (getattr(self, lat), getattr(self, lon)), refs=[ref])
+                    "coordinates", (getattr(self, lat), getattr(self, lon)))
 
     def set_image(self, image_keyword="image"):
         if self.has_non_empty_attribute(image_keyword):
-            self.add_statement("image", getattr(self, image_keyword))
+            self.add_statement("image", getattr(self, image_keyword),
```
Comment (lokal-profil) [inline importer/Monument.py]: I would probably sett `image = getattr(self, image_keyword)` on its own line and then use that in `add_statement` to make it easier to read. (same in a few other places)

File: importer/Monument.py (hunk 2 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

**Comment by lokal-profil** [comment]
Of course there is no need to add docstrings to the methods you know that you are changing/delteing/documenting in #49

File: importer/Monument.py (hunk 3 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 1 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py [hunk 4]
```diff
@@ -17,22 +18,7 @@ def print_wd(self):
...
-        | R1C1
-        | R1C2
-        | R1C3
-        |-
-        | R2C1
-        | R2C2
-        | R2C3
-        |}
-        """
+        """Generate a preview table of the data item."""
```
Comment (lokal-profil) [inline importer/Monument.py]: clarify that it is a wikitext table

File: importer/Monument.py [hunk 5]
```diff
@@ -106,6 +92,16 @@ def add_statement(self, prop_name, value, quals=None, refs=None):
         :param value: the value of the property
         :param quals: a dictionary of qualifiers
         :param refs: a list of references
+
+        Refs is None as default. This means that
```
Comment (lokal-profil) [inline importer/Monument.py]: With a longer text like this I would have it above the parameters.

I would probably add `:param refs: a list of references or a single reference. Set None/True for the default reference, set False to not add a reference.`

Then describe the default above.

File: importer/Monument.py [hunk 6]
```diff
@@ -120,12 +116,18 @@ def add_statement(self, prop_name, value, quals=None, refs=None):
...
             refs = None
+
         if refs and not isinstance(refs, list):
             refs = [refs]
         statement = {"value": value, "quals": qualifiers, "refs": refs}
         base[prop].append(statement)
 
     def remove_statement(self, prop_name):
+        """
+        Remove a statement from the data object.
```
Comment (lokal-profil) [inline importer/Monument.py]: Removes all of them right (if more than one)?

File: importer/Monument.py [hunk 7]
```diff
@@ -148,86 +150,139 @@ def substitute_statement(self, prop_name, value, quals=None, refs=None):
             self.add_statement(prop_name, value, quals, refs)
 
     def set_wd_item(self, wd_item):
+        """Associate the data object with a Q-item."""
```
Comment (lokal-profil) [inline importer/Monument.py]: "Wikidata item"

File: importer/Monument.py [hunk 8]
```diff
@@ -148,86 +150,139 @@ def substitute_statement(self, prop_name, value, quals=None, refs=None):
...
 
     def set_country(self):
+        """Set the country using the mapping file."""
         code = self.mapping["country_code"].lower()
         country = [item["item"]
                    for item in self.adm0 if item["code"].lower() == code][0]
         self.add_statement("country", country)
 
     def set_is(self):
+        """Set the P31 using the mapping file."""
```
Comment (lokal-profil) [inline importer/Monument.py]: "P31 property"

File: importer/Monument.py [hunk 9]
```diff
@@ -148,86 +150,139 @@ def substitute_statement(self, prop_name, value, quals=None, refs=None):
...
         default_is = self.mapping["default_is"]
         self.add_statement("is", default_is["item"])
 
     def set_labels(self, language, content):
+        """
+        Add a label in a specific language using content that potentially
```
Comment (lokal-profil) [inline importer/Monument.py]: First docstring sentence need to fit on one line.

File: importer/Monument.py [hunk 10]
```diff
@@ -148,86 +150,139 @@ def substitute_statement(self, prop_name, value, quals=None, refs=None):
...
+        """
         if self.has_non_empty_attribute(image_keyword):
-            self.add_statement("image", getattr(self, image_keyword),
-                               refs=False)
+            image = getattr(self, image_keyword)
+            self.add_statement("image", image, refs=False)
 
     def set_commonscat(self, keyword="commonscat"):
+        """
+        Add Commons category.
```
Comment (lokal-profil) [inline importer/Monument.py]: Do we want to clarify that we are adding the statement (as compared to `set_registrant_url` which sets an internal variable)?

File: importer/Monument.py [hunk 11]
```diff
@@ -245,21 +300,35 @@ def exists(self, language, article_keyword="monument_article"):
...
     def set_source(self):
+        """Set the 'source' field if present in source data."""
         if self.has_non_empty_attribute("source"):
             self.wd_item["source"] = self.source
 
     def create_stated_in_source(self, source_item, pub_date):
+        """
+        Create a 'stated in' reference.
+
+        :param source_item: Q-item or URL used as a source
```
Comment (lokal-profil) [inline importer/Monument.py]: "Wikidata item" or Qid

Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 9 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 10 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 7 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 8 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 6 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Yes; clarified this.

File: importer/Monument.py (hunk 4 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

File: importer/Monument.py (hunk 5 above)
Comment (Vesihiisi) [inline importer/Monument.py]: Done.

**Pull request merged**